core/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -579,7 +592,7 @@ private void processProblemFactPropertyAnnotation(DescriptorPolicy descriptorPol
             var type = memberAccessor.getType();
             if (!(Collection.class.isAssignableFrom(type) || type.isArray())) {
                 throw new IllegalStateException(
-                        "The solutionClass (%s) has a @%s annotated member (%s) that does not return a %s or an array."
+                        "The solutionClass (%s) has a @%s-annotated member (%s) that does not return a %s or an array."
```

Comment:
This is an unrelated improvement to a fail-fast message which I noticed in a different report.